src/status_im/pairing/core.cljs

```diff
@@ -243,16 +244,22 @@
                   #(when (:public? chat)
                      (models.chat/start-public-chat % (:chat-id chat) {:dont-navigate? true})))))))
 
-(defn handle-pair-installation [{:keys [db] :as cofx} {:keys [installation-id device-type]} timestamp sender]
+(defn handle-pair-installation [{:keys [db] :as cofx} {:keys [name installation-id device-type]} timestamp sender]
   (let [dev-mode? (get-in db [:account/account :dev-mode?])]
     (when (and (config/pairing-enabled? dev-mode?)
                (= sender (accounts.db/current-public-key cofx))
                (not= (get-in db [:account/account :installation-id]) installation-id))
-      (let [installation {:installation-id installation-id
-                          :device-type     device-type
-                          :last-paired     timestamp}]
+      (let [installation {:installation-id   installation-id
+                          :installation-name name
+                          :device-type       device-type
+                          :last-paired       timestamp}]
         (upsert-installation cofx installation)))))
 
+(fx/defn set-name [{:keys [db] :as cofx} installation-name]
+  (let [new-account (assoc (get-in cofx [:db :account/account]) :installation-name installation-name)]
+    {:db (assoc-in db [:account/account :installation-name] installation-name)
```

Comment:
Why don’t we use `new-account` here? `(assoc-in db [:account/account] new-account)`

Reply:
yes, added that code later, best to use new-account, thanks